platform/iphone/gamepad_iphone.mm

```diff
@@ -0,0 +1,290 @@
+#import "gamepad_iphone.h"
+#import "os_iphone.h"
+#import <GameController/GameController.h>
+
+@interface GamepadManager ()
+
+@property(nonatomic, strong) NSMutableDictionary *gamepadById;
+@property(nonatomic, strong) NSMutableArray *gamepads;
+@property(nonatomic, strong) NSMutableArray *pendingGamepads;
+@property(nonatomic, strong) NSMutableArray *availablePlayerIndices;
+
+@end
+
+@implementation GamepadManager
+
++ (instancetype)sharedManager {
+	static dispatch_once_t onceToken;
+	static GamepadManager *gGamepadManager = nil;
+	dispatch_once(&onceToken, ^{
+		gGamepadManager = [[GamepadManager alloc] init];
+	});
+	return gGamepadManager;
+}
+
+- (instancetype)init {
+	self = [super init];
+	if (self) {
+		self.gamepadById = [NSMutableDictionary dictionary];
+		self.gamepads = [NSMutableArray arrayWithCapacity:5];
+		self.pendingGamepads = [NSMutableArray arrayWithCapacity:3];
+		self.availablePlayerIndices = [NSMutableArray arrayWithArray:@[ @(GCControllerPlayerIndex1), @(GCControllerPlayerIndex2), @(GCControllerPlayerIndex3), @(GCControllerPlayerIndex4) ]];
+
+		[[NSNotificationCenter defaultCenter]
+				addObserver:self
+				   selector:@selector(controllerWasConnected:)
+					   name:GCControllerDidConnectNotification
+					 object:nil];
```

Comment:
Just a thought, and perhaps others might disagree with me, but I don't think aligning at the `:` is a good idea readability-wise -- perhaps just align by the beginning of the parameter name?

Reply:
This is _very typical_ for Objective-C code. I've been making use of clang-format for most of these files, perhaps we could add some rules for `.mm` files so we can align on what looks best?

Reply:
I'll leave that up to @akien-mga. I don't personally work with Obj-C code so I can't say what looks best.

Reply:
Looks fine to me, agree its typical for Objective-C. :)

Reply:
👍 Suppose this can be marked as resolved then.